### ISSUE TITLE
Make handling multiple Lua versions easier

### ISSUE THREAD

**Issue by raingloom** [keplerproject/luarocks#444]
I use LuaJiT and 5.3 and aim to be as compatible with anything in between as possible, because of this, I often need multiple versions of modules.

Currently I need three different luarocks installations to suit my workflow, but this is really sub-optimal and makes testing a nightmare.

One solution I found for pure Lua modules is simply symlinking, but this is an ugly workaround and the rocktrees probably won't recognize these versions.

**Comment by xpol** [comment]
I currently use luaenv and setup luarocks for each version.

**Comment by raingloom** [comment]
@xpol  what is your exact workflow? Do you install luarocks from source? Are there any extra configuration changes? How do you handle shell commands, eg.: the ones created by busted or ldoc?

**Comment by xpol** [comment]
Yes, I manually install luarocks form source. and lua and luajit have different header path.

luaenv will hash your luarocks installed shell command such as busted and ldoc.

[Here is luaenv and how it works](https://github.com/cehoffman/luaenv) you may needs [lua-build](https://github.com/cehoffman/lua-build) to make install lua versions easy.

After that install luarocks into `~/.luaenv/versions/\<luaversion\>/` (do NOT install to `~/.luaenv/shims/`)

**Comment by raingloom** [comment]
I already had the Lua install part done (from Arch repos, as they are dependencies for certain packages I use). I guess even if I install the luarocks package I could still override it with luaenv.

So this does, in fact, solve the conflicts, but not the redundancy problem.
I'll see if I can either do some package.path hackery to solve that, which may very well go ugly.
